### ISSUE TITLE
Excessive OR brackets

### ISSUE THREAD

**Issue by jods4** [linq2db/linq2db#2897]
Suppose you have this long chain of `&&`:
```csharp
query.Where(x => x.N > 10 && x.N > 11 && x.N > 12 && x.N > 13 && x.N > 14)
```

Linqtodb generates the following, nice, SQL:
```sql
WHERE
	x.N > 10 AND
	x.N > 11 AND
	x.N > 12 AND
	x.N > 13 AND
	x.N > 14
```

But if you have a chain of `||` such as: 
```csharp
query.Where(x => x.N > 10 || x.N > 11 || x.N > 12 || x.N > 13 || x.N > 14)
```
Then it generates:
```sql
WHERE
	(((x.N > 10 OR x.N > 11) OR (x.N > 12 OR x.N > 13)) OR x.N > 14)
```

Obviously this is still correct, working SQL code, but it's unusually ugly compared to the nice SQL LinqToDb normally generates.

Observe:
- Every pair of OR operands is between brackets, even the top level `WHERE` ones.
- The pairing is strange, it's not left-to-right but looks like a balanced tree.
- It doesn't line-break at operators or even 50 chars. In our real queries this line becomes massively long.

Ideally, all `OR` have the same precedence and this expression could be written out without any bracket, just like ANDs.

**Comment by sdanyliv** [comment]
Agree, will check what happened. EF in this case generates A LOT of parenthesis.

**Comment by MaceWindu** [comment]
99% sure it is caused by condition representation converted to balanced tree by us to avoid stack overflow on large conditions in visitors.

Anyway, SQL builder could be improved to drop parens in more cases
